tensorflow/core/kernels/training_ops.cc

```diff
@@ -2761,7 +2761,8 @@ class SparseApplyFtrlOp : public OpKernel {
     OP_REQUIRES(
         ctx,
         TensorShapeUtils::IsScalar(lr.shape()) &&
-            (lr.scalar<T>()() > static_cast<T>(0) ||
+            (!std::is_same<Device, CPUDevice>::value ||
+             lr.scalar<T>()() > static_cast<T>(0) ||
```

Comment:
On GPU what happens if these constraints are violated?  Do we produce a bogus answer?  Or do crash in the CUDA kernel?

(Please also add the answer as a comment.)

Reply:
The kernel won't crash, just give bogus results. Comment added.

Reply:
The kernel won't crash, just give bogus results.